packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.scss

```diff
@@ -6,216 +6,239 @@
 // Office UI Fabric
 // --------------------------------------------------
 // Checkbox styles
-
-$ms-checkbox-label-size: 20px;
+$ms-checkbox-label-size: 20px !default;
 $ms-checkbox-transition-duration: 200ms;
 $ms-checkbox-transition-timing: cubic-bezier(.4, 0, .23, 1);
-$ms-checkbox-highContrast-iconBoxSize: 16px;
+$ms-checkbox-highContrast-iconBoxSize: 16px !default;
 
-@mixin highContrast-border($active-color, $bw-color, $size: 2px) {
-  @media screen and (-ms-high-contrast: active) {
-    border: $size solid $active-color;
-  }
+// Default state
+.root {
+  overflow: hidden;
+  position: relative;
+  display: flex;
+  align-items: center;
 
-  @media screen and (-ms-high-contrast: black-on-white) {
-    border: $size solid $bw-color;
-  }
-}
+  &:hover,
+  &:active,
+  &:focus {
+    .checkbox {
+      @include ms-borderColor-neutralSecondaryAlt;
+      @media screen and (-ms-high-contrast: active) {
+        border-color: $ms-color-contrastWhiteDisabled;
+      }
 
-@mixin highContrast-color($active-color, $bw-color) {
-  @media screen and (-ms-high-contrast: active) {
-    color: $active-color;
-  }
+      @media screen and (-ms-high-contrast: black-on-white) {
+        border-color: $ms-color-contrastBlackDisabled;
+      }
+    }
+    .text {
+      @include ms-fontColor-black;
+      @media screen and (-ms-high-contrast: active) {
+        color: $ms-color-contrastWhiteDisabled;
+      }
 
-  @media screen and (-ms-high-contrast: black-on-white) {
-    color: $bw-color;
+      @media screen and (-ms-high-contrast: black-on-white) {
+        color: $ms-color-contrastBlackDisabled;
+      }
+    }
+  }
+  &.reversed {
+    > .label {
+        flex-direction: row-reverse;
+    }
   }
 }
 
-.root {
-  box-sizing: border-box;
-  color: $ms-color-neutralPrimary;
-  font-size: $ms-font-size-m;
-  font-weight: $ms-font-weight-regular;
-  min-height: 36px;
-  position: relative;
+.input {
+  margin: 0px;
+  height: 0px;
+  width: 0px;
+  visibility: hidden;
+  opacity: 0;
 }
 
-.textLabel {
-  font-size: $ms-font-size-m;
-  @include padding(0, 0, 0, 26px);
-  display: inline-block;
-  font-size: $ms-font-size-m;
+.label {
+  display: flex;
+  align-items: center;
+  margin: 0 -4px;
+  cursor: pointer;
+
+  > * {
+    margin: 0 4px;
+  }
 }
 
-// The hidden input
-.input {
-  position: absolute;
-  opacity: 0;
-  top: 8px
+.box {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  flex: 0 0 $ms-checkbox-label-size;
+  height: $ms-checkbox-label-size;
 }
 
-// The checkbox square container
-.label::before {
-  content: '';
-  display: inline-block;
-  border: 1px solid $ms-color-neutralTertiary;
-  width: $ms-checkbox-label-size;
+.checkbox {
+  display: flex;
+  align-items: center;
+  justify-content: center;
   height: $ms-checkbox-label-size;
-  font-weight: normal;
-  position: absolute;
+  width: $ms-checkbox-label-size;
+  border: 1px solid black;
```

Comment:
Black is overridden a few lines down where I added `@include ms-borderColor-neutralTertiary;` and it's overrides for high contrast. Changed it to transparent to reduce confusion.